pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -89,14 +95,26 @@ func (c *Controller) finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 			}
 		}
 	}
-	// We wait until all the nodes associated with this nodeClaim have completed their deletion before triggering the finalization of the nodeClaim
-	if len(nodes) > 0 {
-		return reconcile.Result{}, nil
-	}
+	var isInstanceTerminated bool
 	if nodeClaim.Status.ProviderID != "" {
```

Comment:
Comment over this section to describe when the providerID can be set to an empty string (it's when the launch never succeeded or Karpenter never saw the NodeClaim before it deleted)